src/software/ai/navigator/navigator.cpp

```diff
@@ -44,56 +49,54 @@ void Navigator::visit(const KickIntent &kick_intent)
 
 void Navigator::visit(const MoveIntent &move_intent)
 {
+    std::vector<Point> path_points;
+
     Point start =
         this->world.friendlyTeam().getRobotById(move_intent.getRobotId())->position();
     Point dest = move_intent.getDestination();
 
     std::vector<Obstacle> obstacles =
         createCurrentObstacles(move_intent.getAreasToAvoid(), move_intent.getRobotId());
 
-    auto path_planner =
-        std::make_unique<ThetaStarPathPlanner>(this->world.field(), obstacles);
-
-    auto path_points = path_planner->findPath(start, dest);
+    try
+    {
+        path_points = std::get<std::vector<Point>>(
```

Comment:
Should we visit this instead and provide implementation for vector of points and log a msg and stop for `Curve` and `std::monostate` instead of catching `std::bad_variant_access`?

Reply:
so we move navigator stuff into this visitor? would tests have to implement a visitor to test findPath?

Reply:
not sure what the details would look like, i guess the underlying concern is rn we just get the points from the planner and then do a bunch of stuff specific to theta star, how would this extend for spline for example?

Reply:
- We should act under the assumption that the path planner could return points or a polynomials.
- Yes, tests will have to check that the correct variant type is returned from the path planner. 
- In general we should not make assumptions beyond what the interfaces allow.

Reply:
We can do an if else on the type of variant

Reply:
I'm personally a big fan of using this sort of style:
```
        std::visit(overloaded {
            [](auto arg) { std::cout << arg << ' '; },
            [](double arg) { std::cout << std::fixed << arg << ' '; },
            [](const std::string& arg) { std::cout << std::quoted(arg) << ' '; },
        }, v);
```
because it creates clear handlers for each case and I _think_ enforces that there is a handler for every variant type.

Reply:
+1 for `std::visit`

Reply:
there are problems with visit. I'll bring them up in the meeting at 2

Reply:
Done

Reply:
LGTM